server/server.go

```diff
@@ -88,16 +83,23 @@ func NewServer(addr string, opts ...Option) *Server {
 
 	server.handler = &http.Server{
 		Addr:              addr,
-		ReadHeaderTimeout: ReadHeaderTimeout,
-		ReadTimeout:       ReadTimeout,
+		ReadHeaderTimeout: serverConfig.ReadHeaderTimeout,
+		ReadTimeout:       serverConfig.ReadTimeout,
 		BaseContext: func(_ net.Listener) context.Context {
 			return server.baseCtx
 		},
 	}
 
+	server.ApplyServerConfig(serverConfig)
+
 	return server
 }
 
+// Applies the ServerConfig as a value in the baseCtx of the server
+func (s *Server) ApplyServerConfig(serverConfig ServerConfig) {
+	s.baseCtx = context.WithValue(s.baseCtx, reflect.TypeOf(serverConfig), serverConfig)
```

Comment:
please suggest a better key name

Reply:
for context keys the best used some private type that cannot be accessed outside of the package, it'll provide guarantees that it'll never have any conflicts. here example:

```golang

type serverCtxKey struct{}


func (s *Server) setServerConfig(ctx context.Context, config ServerConfig) {
    s.baseCtx = context.WithValue(s.baseCtx, serverCtxKey{}, serverConfig)
}

func GetServerConfig(ctx) ServerConfig {
    config, ok := ctx.Value(serverCtxKey{})
    if !ok {
        return defaultServerConfig
    }

    return config
} 

```

Reply:
thanks added